docs/enhanced-provider-k8s.md

```diff
@@ -0,0 +1,314 @@
+* Create and update are usually coupled together.
+
+We only need 4 types of management policies:
+
+| Observe | Create & Update | Delete | Management Policy   | Memo
+|:--------|:----------------|:-------|:--------------------|:----
+| Y       | Y               | Y      | Default             | The provider can fully manage the resource. This is the default policy.
+| Y       | Y               | N      | ObserveCreateUpdate | The provider can observe, create, or update the resource, but can not delete it. 
+| Y       | N               | Y      | ObserveDelete       | The provider can observe or delete the resource, but can not create and update it.
+| Y       | N               | N      | Observe             | The provider can only observe the resource. This maps to the read-only scenario where the resource is fully controlled by third party application and the provider only reads the status of the resource.
+
+The idea of Resource Management Policy is actually expanded from the concept of `DeletionPolicy` in Crossplane which can be found in Crossplane code:
+
+```go
+const (
+    // DeletionOrphan means the external resource will orphaned when its managed
+    // resource is deleted.
+    DeletionOrphan DeletionPolicy = "Orphan"
+
+    // DeletionDelete means both the external resource will be deleted when its
+    // managed resource is deleted.
+    DeletionDelete DeletionPolicy = "Delete"
+)
+```
+
+The `DeletionOrphan` policy is actually equivalent to the `ObserveCreateUpdate` resource management policy.
+
+Please note that currently the resource management policy and DeletionPolicy are two separate mechanisms to declare the resource policy at different level: Crossplane runtime level vs. provider level. To avoid the potential conflict or confusion, it is highly recommended to use the policy at provider level over Crossplane runtime level when you use Provider Kubernetes as the provider level policy is a super set of Crossplane runtime level policy and has more options.
+
+In case both are defined, to determine if the resource can be deleted or not, both policies need to be considered, and the one at Crossplane runtime level will be taken into account at first. That means:
+
+* If `DeletionPolicy` is defined, then the resource will not be deleted by provider, no matter what policy is defined at provider level.
+* If `DeletionDelete` is defined, then relying on the policy defined at provider level, if `ObserveCreateUpdate` or `Observe` is specified, then the resource will not be deleted, otherwise, it will be deleted by the provider.
+
+Now, let's define policy for our `Object` resource that represents the `ClusterServiceVersion` resource. In our case, we should use `ObserveDelete` policy. Because the policies in the enhanced provider are implemented as Kubernetes annotations, you just need to specify the policy by adding an annotation to the `Object` resource. For example:
```

Comment:
I feel like going with an explicit spec field would be more clear and be consistent with how crossplane defines `DeletionPolicy`. What do you think about using a `spec.managementPolicy` instead of annotation?

Reply:
Agreed, updated.